packages/shared/src/components/cards/common/list/ActionButtons.tsx

```diff
@@ -59,9 +62,31 @@ export default function ActionButtons({
       onClose(true);
     }
 
+    if (post.userState?.vote !== UserVote.Up) {
+      onInteract('upvote');
+    }
+    if (interaction === 'upvote') {
+      onInteract();
+    }
+
     onUpvoteClick?.(post);
   };
 
+  const onToggleBookmark = () => {
+    if (!post.bookmarked) {
+      onInteract('bookmark');
+    }
+    if (interaction === 'bookmark') {
+      onInteract();
+    }
+    onBookmarkClick(post);
+  };
+
+  const onCopyLink = (e: React.MouseEvent) => {
+    onInteract('copy');
+    onCopyLinkClick?.(e, post);
+  };
```

Comment:
I like the direction of implementing it in one place but then again we have this if conditions which if changed can again screw up the flow. I would expect if we are standardizing it that the hook would provide ready made interactions, or specialized sub hooks for interaction types.